### PR TITLE
Feature/image size

### PR DESCRIPTION
**Pull request by lbebber**
#### What is the purpose of this pull request?
Gets images of varied sizes based on the screen size.

To test, open the following url in varied screen sizes and watch the network tab filtered by images, to see which images are being loaded.
https://lbebber--alssports.myvtex.com/

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by lbebber** [comment]
@Klynger working on it right now

**Comment by klzns** [comment]
> @Klynger working on it right now

ops hehe

File: react/Banner.tsx [hunk 1]
```diff
@@ -61,8 +71,26 @@ const Banner = (props: Props) => {
...
+      ? mobileImage
+      : image
+
   const handles = useCssHandles(CSS_HANDLES)
 
+  const srcSet = useMemo(
+    () =>
+      isImageFromFileManager(src)
+        ? IMAGE_SIZES.map(size => `${resizeImage(src, size)} ${size}w`).join(
+            ','
```
Comment (lbebber) [inline react/Banner.tsx:88]: terrivel essa formataçao

Comment (klzns) [reply react/Banner.tsx:88]: caos

**Comment by lbebber** [comment]
@klzns It has to do with how "looping" works on this component (it's loading the same two images twice, the browser smartly cancels one of the requests).

It has nothing to do with this PR, the same is happening on prod

File: react/Carousel.tsx [hunk 2]
```diff
@@ -146,7 +147,13 @@ const Carousel = (props: Props) => {
             style={{ maxHeight: height }}
             sliderTransitionDuration={500}
           >
-            <Banner height={height} {...banner} />
+            {i > 1 ? (
```
Comment (iaronaraujo) [inline react/Carousel.tsx:150]: Is this check correct? Shouldn't it be `0` or `>=` ?

File: react/Banner.tsx (hunk 1 above)
Comment (kaisermann) [reply react/Banner.tsx:88]: Adiciona uns `{}` nessa arrow function pra melhorar a legibilidade disso :harold-pain:

File: react/Banner.tsx [hunk 3]
```diff
@@ -61,8 +71,26 @@ const Banner = (props: Props) => {
...
+          )
+        : undefined,
+    [src]
+  )
```
Comment (kaisermann) [inline react/Banner.tsx:92]:
```suggestion
const srcSet = useMemo(() => {
  if (isImageFromFileManager(src)) {
    return IMAGE_SIZES.map(size => `${resizeImage(src, size)} ${size}w`).join(',')
  }
  return undefined
}, [src])
```

File: react/Carousel.tsx (hunk 2 above)
Comment (lbebber) [reply react/Carousel.tsx:150]: @iaronaraujo good catch 👌

File: react/Banner.tsx (hunk 3 above)
Comment (lbebber) [reply react/Banner.tsx:92]: @kaisermann boa sugestao, mas essa logica vai ser feita em outro componente mesmo

File: react/Banner.tsx (hunk 1 above)
Comment (klzns) [reply react/Banner.tsx:88]: Ou coloca a funcao do map definida fora

**Comment by victorhmp** [comment]
@lbebber Does this still make sense? Or can we close it?

File: react/Banner.tsx [hunk 4]
```diff
@@ -61,8 +71,26 @@ const Banner = (props: Props) => {
   } = props
 
   const { isMobile, device } = useDevice()
+
+  const src =
+    device === 'tablet'
+      ? tabletImage || image
+      : isMobile && mobileImage
+      ? mobileImage
+      : image
```
Comment (kaisermann) [inline react/Banner.tsx:80]:
```suggestion
  const src = (device === 'tablet' ? tabletImage : isMobile ? mobileImage : image) ?? image
```

**Comment by lbebber** [comment]
a bit abandoned ATM, might be revived if the need arises

**Pull request closed without merging**